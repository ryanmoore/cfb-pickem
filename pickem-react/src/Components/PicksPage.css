.picklist-left {
    float: right;
}

.picklist-left th.team {
    text-align: right;
    border: none;
    padding: 0;
}

.picklist-right {
    float: left;
}

.picklist-right th.team {
    text-align: left;
    border: none;
    padding: 0;
}

td.wager {
    width: 20px;
    text-align: right;
    font-weight: bold;
}

td.user {
    width: auto;
}

.picklist-left td.user {
    text-align: right;
    border: none;
    padding: 0;
}

.picklist-left td.wager{
    text-align: left;
    padding: 0px 0px 0px 5px;
    border: none;
}

.picklist-right td.user {
    text-align: left;
    border: none;
    padding: 0;
}

.picklist-right td.wager{
    text-align: right;
    padding: 0px 5px 0px 0px;
    border: none;
}

.game-details {
    text-align: center;
}

.matchup-row:nth-of-type(odd) {
    background-color: #f4f4f4;
}

.matchup-row {
    border-top: 1px solid #ddd;
    padding-top: 4px;
    padding-bottom: 20px;
}

.winning-pick table {
    color: #008B00;
}

.losing-pick table {
    color: #E3170D;
}
