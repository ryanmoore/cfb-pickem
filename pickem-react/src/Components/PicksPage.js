import React, {
    Component
} from 'react';
import {
    Row,
    Col,
    Table,
    Grid
} from 'react-bootstrap';
import './PicksPage.css';
import moment from 'moment';


class PicksCol extends Component {
    static propTypes = {
        picks: React.PropTypes.arrayOf(
            React.PropTypes.shape({
                username: React.PropTypes.string.isRequired,
                wager: React.PropTypes.number.isRequired,
            })),
        teamName: React.PropTypes.string.isRequired,
        left: React.PropTypes.bool.isRequired,
        winner: React.PropTypes.bool.isRequired,
        decided: React.PropTypes.bool.isRequired,
        admin: React.PropTypes.bool,
        AdminButton: React.PropTypes.node,
        participantId: React.PropTypes.number.isRequired,
    }
    render() {
        const {
            picks,
            teamName,
            left,
            winner,
            decided,
            admin,
            AdminButton,
            participantId,
        } = this.props;
        const sortByWagerDesc = (a, b) => {
            if(a.wager < b.wager) return 1;
            if(a.wager > b.wager) return -1;
            if(a.username < b.username) return 1;
            if(a.username > b.username) return -1;
            return 0;
        };
        const colClass = winner ? 'winning-pick' : decided ? 'losing-pick' : '';
        const tableClass = 'picklist-' + (left ? 'left' : 'right');
        var sortedPicks = picks.slice(0);
        sortedPicks.sort(sortByWagerDesc)
        const tableRows = sortedPicks.map((pick) => {
            const wager = <td className='wager'>{ pick.wager }</td>;
            const user = <td className='user'>{ pick.username }</td>;
            if (left) {
                return <tr key={pick.username}>{ user }{ wager }</tr>
            } else {
                return <tr key={pick.username}>{ wager }{ user }</tr>
            }
        });
        return (<Col xs={4} className={colClass}>
            <Table className={tableClass}>
                <thead>
                    <tr>
                        <th className='team' colSpan={2}>{ teamName }</th>
                    </tr>
                </thead>
                <tbody>
                    { tableRows }
                </tbody>
            </Table>
            { admin && <AdminButton id={participantId}
                            left={left}
                            decided={decided}/> }
        </Col>);
    }
}

class GameDetail extends Component {
    static propTypes = {
        name: React.PropTypes.string.isRequired,
        date: React.PropTypes.string.isRequired,
    };
    render() {
        const {
            name,
            date
        } = this.props;
        return (
            <Col xs={4} className='game-details'>
                <p>{ name }<br/>{ date }</p>
            </Col>
        );
    }
}

class PicksRow extends Component {
    static propTypes = {
        left: React.PropTypes.shape({
            picks: React.PropTypes.array.isRequired,
            teamName: React.PropTypes.string.isRequired,
        }).isRequired,
        right: React.PropTypes.shape({
            picks: React.PropTypes.array.isRequired,
            teamName: React.PropTypes.string.isRequired,
        }).isRequired,
        gameDetails: React.PropTypes.shape({
            eventName: React.PropTypes.string.isRequired,
            date: React.PropTypes.instanceOf(Date).isRequired,
        }).isRequired,
        winner: React.PropTypes.number,
        admin: React.PropTypes.bool,
        AdminButton: React.PropTypes.node,
    }
    render() {
        const {
            left,
            right,
            gameDetails,
            winner,
            admin,
            AdminButton,
        } = this.props;
        const createPickCol = (pickdata, isLeft) => {
            return <PicksCol
                participantId={pickdata.id}
                picks={ pickdata.picks }
                teamName={ pickdata.teamName }
                left={isLeft}
                winner={pickdata.id === winner}
                decided={!!winner}
                admin={admin}
                AdminButton={AdminButton}
            />
        }
        return (
            <Row className='matchup-row'>
                { createPickCol(left, true) }
                <GameDetail 
                    name={ gameDetails.eventName }
                    date={ moment(gameDetails.date).format('lll') }
                />
                { createPickCol(right, false) }
            </Row>
        );
    }
}

class PicksPage extends Component {
    static propTypes = {
        matchupPicks: React.PropTypes.arrayOf(React.PropTypes.shape({
            left: React.PropTypes.any.isRequired,
            right: React.PropTypes.any.isRequired,
            gameDetails: React.PropTypes.any.isRequired,
            id: React.PropTypes.number.isRequired,
            winner: React.PropTypes.number,
        })).isRequired,
        admin: React.PropTypes.bool,
        AdminButton: React.PropTypes.node,
    };
    render() {
        const {
            matchupPicks,
            admin,
            AdminButton,
        } = this.props;
        const createRow = (picks) => {
            return <PicksRow key={picks.id}
                left={picks.left}
                right={picks.right}
                gameDetails={picks.gameDetails}
                winner={picks.winner}
                admin={admin}
                AdminButton={AdminButton}
                />
        }
        const undecided = matchupPicks.filter((elt) => !elt.winner).map(createRow);
        const decided = matchupPicks.filter((elt) => !!elt.winner).map(createRow);
        return (
            <Grid>
                {undecided}
                <h1>Completed Games</h1>
                {decided}
            </Grid>
        );
    }
}

export default PicksPage;
