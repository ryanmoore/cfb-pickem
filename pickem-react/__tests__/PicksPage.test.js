import React from 'react';
import PicksPage from '../src/Components/PicksPage';
import renderer from 'react-test-renderer';

describe('PicksPage', () => {
    it('should render sample data', () => {
        const sampleData = [{
            left: {
                picks: [{username: 'Ryan', wager: 1}],
                teamName: 'Left1'
            },
            right: {
                picks: [{username: 'Dan', wager: 2}],
                teamName: 'Right1'
            },
            gameDetails: {
                eventName: 'Game1',
                date: new Date(2016, 11, 1, 13, 0, 0),
            },
            id: 1,
        }];
        const component = renderer.create(
            <PicksPage matchupPicks={sampleData} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
