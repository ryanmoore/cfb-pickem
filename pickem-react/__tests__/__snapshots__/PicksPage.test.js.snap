exports[`PicksPage should render sample data 1`] = `
<div
  className="container">
  <div
    className="matchup-row row">
    <div
      className="winning-pick col-xs-4">
      <table
        className="picklist-left table">
        <thead>
          <tr>
            <th
              className="team"
              colSpan={2}>
              Left1
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="user">
              Ryan
            </td>
            <td
              className="wager">
              1
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="game-details col-xs-4">
      <p>
        Game1
        <br />
        Dec 1, 2016 1:00 PM
      </p>
    </div>
    <div
      className="winning-pick col-xs-4">
      <table
        className="picklist-right table">
        <thead>
          <tr>
            <th
              className="team"
              colSpan={2}>
              Right1
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="wager">
              2
            </td>
            <td
              className="user">
              Dan
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <h1>
    Completed Games
  </h1>
</div>
`;
